workflow/systems/base/adaptor.py

```diff
@@ -0,0 +1,34 @@
+from abc import ABC, abstractmethod
+from typing import Dict
+from django.utils.functional import classproperty
+
+
+class WorkflowAdaptor(ABC):
+    """
+    Abstract class for implementing specific workflow system integration.
+    """
+
+    @classmethod
+    @classproperty
+    @abstractmethod
+    def system(cls) -> str:
+        """
+            Property describing a keyword to refer to this workflow system
+        """
+        pass
+
+    @classmethod
+    @abstractmethod
+    def get_groups(cls) -> Dict:
+        """
+             Get all available groups from this workflow system as result dict
+        """
+        pass
+
+    @classmethod
+    @abstractmethod
+    def get_workflows(cls, name: str = None, group: str = None) -> Dict:
```

Comment:
I made it dependent on the handler just because it's hard to know what the system will return. Possible future improvement for this is to store the result in the handler and create an interface to retrieve information like status, error messages or output payload.